### DISCUSSION TITLE
Visualizer Unable Function

### DISCUSSION THREAD

**Discussion by Skies034** [mpurses/Sonder#26] [Q&A] [answered]
I have 2 pc (Desktop and Laptop) 

Desktop: Visualizer able to function normally.
Laptop: Visualizer unable to function.

both is latest version of win10 pro.
both rainmeter have the same setting and version.
both audio output is same (Realtek High Definition Audio)

**Comment by mpurses** [comment, marked as answer]
Could you try going through the troubleshooting list to see if it fixes anything: Most likely it is the Sensitivity or dot size.

- Increase the Audio 'Sensitivity' setting(try increasing up to 100), your music may be too quiet or your system isn't registering at lower dB for the audio device.
- Increase Dot Size - may be too small for your monitor.

**Reply by Skies034** [reply]
I have all tried, unable show too.
All setting changed to max (incl volume).

will the graphic card affect the Visualizer ?
Desktop: single card; K620
Laptop: dual card; Intel 4600 and GTX 860M - Screen out put is Intel 4600

**Reply by Skies034** [reply]
The log files error:
Script: DotGenerator_Stargazer_wanderer.lua:77: attempt to index global 'audioMeasure' (a nil value) 
Script: MeasureGenerator.lua:6: attempt to index local 'file' (a nil value)

**Reply by mpurses** [reply]
Could you download this version: https://github.com/mpurses/Sonder/raw/master/Pre-Releases/Sonder_2.11.0_PreRelease.rmskin 

I added some debug logging to the Wanderer variant (so just test with this one going forward).  It should look something like this in the Log:
![Screenshot 2021-08-16 092518](https://user-images.githubusercontent.com/642218/129621704-3f26478f-d47f-44c6-935e-22bb4f751e86.png)

Yours will probably fail somewhere, so hopefully it will tell us something...

**Reply by mpurses** [reply]
Also in "Documents\Rainmeter\Skins\Sonder\@Resources\Visualizer" post the contents of **MeasureAudio.inc**.  I want to make sure it is writing that file correctly.

**Reply by Skies034** [reply]
> Could you download this version: https://github.com/mpurses/Sonder/raw/master/Pre-Releases/Sonder_2.11.0_PreRelease.rmskin
> 
> I added some debug logging to the Wanderer variant (so just test with this one going forward). It should look something like this in the Log:
> ![Screenshot 2021-08-16 092518](https://user-images.githubusercontent.com/642218/129621704-3f26478f-d47f-44c6-935e-22bb4f751e86.png)
> 
> Yours will probably fail somewhere, so hopefully it will tell us something...

After i reinstall Sonder_2.11.0, it works perfectly for me. The Virtualizer able to function normally.  Thanks a lot

**Reply by Skies034** [reply]
> Also in "Documents\Rainmeter\Skins\Sonder@Resources\Visualizer" post the contents of **MeasureAudio.inc**. I want to make sure it is writing that file correctly.

![image](https://user-images.githubusercontent.com/87711849/131239585-4031262c-91be-4fb8-b4df-9c6a7f200bb1.png)

It Failed to write.

**Reply by mpurses** [reply]
Okay, that's what I expected. It's having trouble with the non-english characters for the username.  You may run into some editing issues when selecting different variations. The MeasureGenerator.lua edits the MeasureAudio.inc file on the fly based on the _Width_ or _Peak_ of the Visualizer.  But it will be hard to notice the actual "errors" in the Visualizer. That's why it's "working" but adjusting it outside of the default values for Width/Peak will fail silently now. I know the line one won't work based on not being able to write to that file though. 

I'll try to work on fixing this though.
